mmdet3d/apis/inference.py

```diff
@@ -155,23 +155,25 @@ def inference_multi_modality_detector(model, pcd, image, ann_file):
         bbox_fields=[],
         mask_fields=[],
         seg_fields=[])
-
-    # depth map points to image conversion
-    if box_mode_3d == Box3DMode.DEPTH:
-        data.update(dict(calib=info['calib']))
-
     data = test_pipeline(data)
 
+    # TODO: this code is dangerous and dataset-specific. Move lidar2img and
+    #       depth2img to .pkl annotations in the future.
```

Comment:
I am a little confused why you think this is dangerous? Are there any cases where `lidar2img` or `depth2img` is not calculated like this?

Reply:
The thing is that all this function is dataset-specific, however it has a general name `inference_multi_modality_detector`. Such things as `R0_rect`, `Tr_velo_to_cam` are KITTI-specific. Others like `info['calib']['Rt']` are SUN RGN-D-specific. What if we have for example nuScenes multi-modality detector? I think if we want to keep this API, it can be refactored in the future. For example by putting `lidar2img` and `depth2img` to `.pkl` annotations preliminarily. For sure can remove this `todo`, if you disagree.

Reply:
I agree with you. Let's keep this `TODO`.